### PR TITLE
Fix problems reported by @dzoep.

### PR DESCRIPTION
**Pull request by samth**

### CODE REVIEW

File: typed-racket-lib/typed-racket/base-env/base-env.rkt [hunk 1]
```diff
@@ -1085,6 +1089,10 @@
 [make-immutable-custom-hash (->opt (-> Univ Univ Univ) (-> Univ -Nat) [(-> Univ -Nat)] Univ)]
 [make-weak-custom-hash (->opt (-> Univ Univ Univ) (-> Univ -Nat) [(-> Univ -Nat)] Univ)]
 
+
+[hash-union (-poly (a b) (->* (list (-Immutable-HT a b))  (-HT a b) (-Immutable-HT a b)))]
+[hash-union! (-poly (a b) (->* (list (-Mutable-HT a b))  (-HT a b) -Void))]
```
Comment (sorawee) [inline typed-racket-lib/typed-racket/base-env/base-env.rkt:1094]: `hash-intersect` too?

Comment (samth) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt:1094]: This was just aiming at #1025, which is just those two functions.

Comment (samth) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt:1094]: In fact, there is no `hash-intersect`

Comment (sorawee) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt:1094]: See https://github.com/racket/racket/pull/3310

Comment (samth) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt:1094]: Ah, that's what I get for looking at docs.racket-lang.org. Unfortunately there's a typo in the docs ...

**Pull request merged**